challenge/dns01/nameserver.go

```diff
@@ -51,9 +48,7 @@ func (cache *soaCacheEntry) isExpired() bool {
 
 // ClearFqdnCache clears the cache of fqdn to zone mappings. Primarily used in testing.
 func ClearFqdnCache() {
-	muFqdnSoaCache.Lock()
-	fqdnSoaCache = map[string]*soaCacheEntry{}
-	muFqdnSoaCache.Unlock()
+	fqdnSoaCache.Clear()
```

Comment:
`Clear()` was added in Go 1.23, please update go.mod or use API compatible with Go 1.22. Thank you

Reply:
We will not update to go1.23, but I will change this.